docs/SPEC.md

```diff
@@ -788,6 +795,14 @@ Array literals construct a value with the array type.
     ArrayLiteral   = "[" ExpressionList "]" .
     ExpressionList = [ Expression { "," Expression } ] .
 
+##### Dictionary literals
+
+Dictionary literals construct a value with the dict type.
+
+    DictLiteral     = "[" AssociativeList "]" .
+    AssociativeList = [ Association { "," AssociativeList } ] .
+    Association     = Expression ":" Expression .
```

Comment:
Is this ambiguous with array literals? The  `Association = `Expression ":" Expression`  production rule and this one `ExpressionList = [ Expression { "," Expression } ] .` We would need infinite back tracking to wait for either the `:` or the `,`.

Reply:
There's no ambiguity except in the case of an empty array/dict `[]`. However you are correct that it would require infinite backtracking in order to determine the type. I believe our parser is LL(1) so this syntax might not work (assuming we keep keys as arbitrary expressions)